ClimsoftVer4/ClimsoftVer4/ucrDay.vb

```diff
@@ -1,6 +1,8 @@
 ﻿Public Class ucrDay
     'Private strDaysTableName As String = "days"
     Private strDay As String = "day"
+    'TODO
+    'Is it necessary to have 4 datatables for this control?
     Private dtb28 As DataTable
     Private dtb29 As DataTable
     Private dtb30 As DataTable
```

Comment:
@maxwellfundi @dannyparsons  the 3 datatables could be replaced by implementing the 2 subroutines as shown below

```
Public Sub InitialiseControl()
        dtbRecords = New DataTable
        dtbRecords.Columns.Add(strDay, GetType(Integer))
    End Sub
    Public Overrides Sub PopulateControl()
        'MyBase.PopulateControl()
        Dim lstShortMonths As New List(Of String)({4, 6, 9, 11})
        Dim iMonth As Integer
        Dim iNumOfDays As Integer

        'get the correct number of days
        iMonth = If(ucrLinkedMonth Is Nothing, 1, ucrLinkedMonth.GetValue)
        If iMonth = 2 Then
            iNumOfDays = If(ucrLinkedYear Is Nothing, 29, (If(DateTime.IsLeapYear(ucrLinkedYear.GetValue), 29, 28)))
        Else
            iNumOfDays = If(lstShortMonths.Contains(iMonth), 30, 31)
        End If

        'set up the datatable. 
        'we redo this here to prevent errors that maybe cause during concurrency in GetValue
        InitialiseControl()

        'fill in the days to show
        For i As Integer = 1 To iNumOfDays
            dtbRecords.Rows.Add(i)
        Next

        dtbRecords.DefaultView.Sort = strDay & " ASC"
            If dtbRecords.Rows.Count > 0 Then
                cboValues.DataSource = dtbRecords
                cboValues.ValueMember = strDay
                If bFirstLoad Then
                    SetViewTypeAsDay()
                End If
            Else
                cboValues.DataSource = Nothing
            End If

    End Sub
```

I didn't implement this because I'm not sure why currently it's implemented the way it is. Any suggestion or comment is welcome?

Reply:
@dannyparsons please go through this